data-prepper-plugins/s3-source/src/integrationTest/java/com/amazon/dataprepper/plugins/source/SqsWorkerIT.java

```diff
@@ -0,0 +1,116 @@
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.buffer.Buffer;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.record.Record;
+import com.amazon.dataprepper.plugins.source.codec.Codec;
+import com.amazon.dataprepper.plugins.source.configuration.OnErrorOption;
+import com.amazon.dataprepper.plugins.source.configuration.SqsOptions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.sqs.SqsClient;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SqsWorkerIT {
+    private S3Client s3Client;
+    private SqsClient sqsClient;
+    private S3Service s3Service;
+    private Buffer<Record<Event>> buffer;
+    private S3SourceConfig s3SourceConfig;
+    private S3ObjectGenerator s3ObjectGenerator;
+    private String bucket;
+    private Codec codec;
+    private PluginMetrics pluginMetrics;
+
+    @BeforeEach
+    void setUp() {
+        s3Client = S3Client.builder()
+                .region(Region.of(System.getProperty("tests.s3source.region")))
+                .build();
+        bucket = System.getProperty("tests.s3source.bucket");
+        s3ObjectGenerator = new S3ObjectGenerator(s3Client, bucket);
+
+
+        sqsClient = SqsClient.builder()
+                .region(Region.of(System.getProperty("tests.s3source.region")))
+                .build();
+
+        s3SourceConfig = mock(S3SourceConfig.class);
+        buffer = mock(Buffer.class);
+        codec = mock(Codec.class);
+        pluginMetrics = mock(PluginMetrics.class);
+        s3Service = mock(S3Service.class);
+
+        final SqsOptions sqsOptions = mock(SqsOptions.class);
+        when(sqsOptions.getSqsUrl()).thenReturn(System.getProperty("tests.s3source.queue.url"));
+        when(sqsOptions.getVisibilityTimeout()).thenReturn(Duration.ofSeconds(60));
+        when(sqsOptions.getMaximumMessages()).thenReturn(10);
+        when(sqsOptions.getWaitTime()).thenReturn(Duration.ofSeconds(10));
+        when(s3SourceConfig.getSqsOptions()).thenReturn(sqsOptions);
+        when(s3SourceConfig.getOnErrorOption()).thenReturn(OnErrorOption.DELETE_MESSAGES);
+    }
+
+    private SqsWorker createObjectUnderTest() {
+        return new SqsWorker(sqsClient, s3Service, s3SourceConfig);
+    }
+
+    @AfterEach
+    void processRemainingMessages() {
```

Comment:
Are we potentially masking issues with the tests? Will this impact other tests? What happens if two integration tests run at the same time?

Reply:
I'm not sure if I understood, but it won't impact other tests. In future if we make this a GitHub action and run two tests at same time with same S3 bucket and SQS it might impact.

Reply:
It would be possible to use `PurgeQueue`. This does not solve the simultaneous test runs however.

https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_PurgeQueue.html

In general, I'm fine with deferring the solution on that until we do have a GHA. Currently, these tests are run only by individuals on their own AWS accounts.